stdlib/number.gr

```diff
@@ -660,3 +660,89 @@ provide let toRadians = degrees => degrees * (pi / 180)
  * @since v0.5.4
  */
 provide let toDegrees = radians => radians * (180 / pi)
+
+/**
+ * Clamps a number to a given range.
+ *
+ * @param range: The inclusive range to clamp within.
+ * @param input: The number to clamp
+ * @returns The clamped number
+ *
+ * @since v0.6.0
+ */
+provide let clamp = (range, input) => {
+  if (isNaN(input)) {
+    input
+  } else {
+    let rangeEnd = max(range.rangeStart, range.rangeEnd)
+    let rangeStart = min(range.rangeStart, range.rangeEnd)
+
+    if (input > rangeEnd) rangeEnd else if (input < rangeStart) rangeStart
+    else input
+  }
+}
+
+/**
+ * Linearly interpolates across the provided range by the given amount.
+ *
+ * @param range: An inclusive Range to interpolate between
+ * @param amount: an amount between 0 and 1, anything outside the range will produce an extrapolation, i.e. `lerp(0, 100, 2)` will produce 200
+ * @returns The interpolated value
+ *
+ * @throws InvalidArgument(String): When `amount` is not between 0 and 1 or `range` is not finite
```

Comment:
These need to be separate throws entries. graindoc joins them into bullets so you should specify each separately